controller/src/main/java/io/pravega/controller/store/stream/BucketStore.java

```diff
@@ -0,0 +1,129 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.controller.store.stream;
+
+import io.pravega.controller.server.bucket.BucketChangeListener;
+import io.pravega.controller.server.bucket.BucketOwnershipListener;
+import lombok.Getter;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+
+/**
+ * Stream Metadata.
+ */
+public interface BucketStore {
+    /**
+     * Method to get count of buckets in the store.
+     *
+     * @return number of buckets.
+     */
+    int getBucketCount();
+
+    /**
+     * Method to register listener for changes to bucket's ownership.
+     *
+     * @param serviceType service type
+     * @param listener    listener
+     */
+    void registerBucketOwnershipListener(ServiceType serviceType, BucketOwnershipListener listener);
```

Comment:
We should be using inversion of control. Let's find a way to pass in the listener or a dispatcher into the bucket store at the time of it's construction.